src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -11580,129 +11753,812 @@ third
+        System.Console.WriteLine(d)
+    End Sub
+End Class
+
+Class A
+    Public Shared Widening Operator CType(val As String) As A
+        System.Console.WriteLine(val)
+        Return New A()
+    End Operator
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe, additionalRefs:=s_valueTupleRefs)
+
+            CompileAndVerify(comp, expectedOutput:="
+1
+1
+1
+1
+(A, 1)
+(A, 1)
+(A, 1)
+(A, 1)")
+        End Sub
+
+        <Fact>
+        Public Sub UserDefinedConversions_02()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Option Strict Off
+
+Public Class C
+    Shared Sub Main()
+        Dim int = 1
+        Dim val as new B()
+        Dim tuple = (val, int)
+        Dim a as (Integer, Integer) = tuple
+        Dim b as (Integer, Integer) = (val, int)
+        Dim c as (Integer, Integer)? = tuple
+        Dim d as (Integer, Integer)? = (val, int)
+
+        System.Console.WriteLine(a)
+        System.Console.WriteLine(b)
+        System.Console.WriteLine(c)
+        System.Console.WriteLine(d)
+    End Sub
+End Class
+
+Class B
+    Public Shared Widening Operator CType(val As B) As String
+        System.Console.WriteLine(val Is Nothing)
+        Return "2"
+    End Operator
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe, additionalRefs:=s_valueTupleRefs)
+
+            CompileAndVerify(comp, expectedOutput:="
+False
+False
+False
+False
+(2, 1)
+(2, 1)
+(2, 1)
+(2, 1)")
+        End Sub
+
+        <Fact>
+        <WorkItem(14530, "https://github.com/dotnet/roslyn/issues/14530")>
+        Public Sub UserDefinedConversions_03()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb">
+Option Strict Off
+
+Public Class C
+    Shared Sub Main()
+        Dim int = 1
+        Dim ad = Function() 2
+        Dim tuple = (ad, int)
+        Dim a as (A, Integer) = tuple
+        Dim b as (A, Integer) = (ad, int)
+        Dim c as (A, Integer)? = tuple
+        Dim d as (A, Integer)? = (ad, int)
+
+        System.Console.WriteLine(a)
+        System.Console.WriteLine(b)
+        System.Console.WriteLine(c)
+        System.Console.WriteLine(d)
+    End Sub
+End Class
+
+Class A
+    Public Shared Widening Operator CType(val As System.Func(Of Integer, Integer)) As A
+        System.Console.WriteLine(val)
+        Return New A()
+    End Operator
+End Class
+    </file>
+</compilation>,
+options:=TestOptions.ReleaseExe, additionalRefs:=s_valueTupleRefs)
+
+            CompileAndVerify(comp, expectedOutput:="
+System.Func`2[System.Int32,System.Int32]
+System.Func`2[System.Int32,System.Int32]
+System.Func`2[System.Int32,System.Int32]
+System.Func`2[System.Int32,System.Int32]
+(A, 1)
+(A, 1)
+(A, 1)
+(A, 1)")
+        End Sub
+
+        <Fact>
+        Public Sub Inference02_Addressof()
+
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation name="Tuples">
```

Comment:
There is no need and I don't believe I added the name. it looks like an invalid merge, Julian's change for removal of the name got matched to the test I added before merging his change.